File: Corretora.imoveis.Core/Models/Imovel/Imovel.cs

using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Corretora.Imoveis.Core.Models.Imovel
{
    public class Imovel
    {
        public long IdImovel { get; set; }
        public string Tipo { get; set; }
        public string Descricao { get; set; }
        public DateTime DataPlubicacao { get; set; }
        [DataType(DataType.Currency)]
        public decimal ValorImovel { get; set; }
        [DataType(DataType.Currency)]
        public decimal ValorEstimadoDocumentacao { get; set; }
        public string AceitaFinanciamento { get; set; }
        public string Observacao { get; set; }
        public string Quarto { get; set; }
        public string Banheiro { get; set; }
        public string Garagem { get; set; }
        public string AreaUtil { get; set; }
        public string Cep { get; set; }      
        public string Logadouro { get; set; }
        public string Numero { get; set; }

        public string Bairro { get; set; }
        public string Cidade { get; set; }

        public string UF { get; set; }

        public decimal ValorDe { get; set; }
        public decimal ValorAte { get; set; }

      

        public Imovel()
        {
            Imagens = new List<AnexoImovel>();

            ValorImovel = ValorDe;
            ValorImovel = ValorAte;
           
        }

        public ICollection<AnexoImovel> Imagens { get; set; }



        public Imovel(DataRow dr)
        {
           
            Logadouro = dr["logadouro"].ToString();
            Numero = dr["numero"].ToString();
            Cidade = dr["cidade"].ToString();
            UF = dr["uf"].ToString();
            Cep = dr["cep"].ToString();
            Bairro = dr["bairro"].ToString();

        }


        public void InformarAnexo(AnexoImovel anexoImovel)
        {
            if (Imagens == null)
                Imagens = new List<AnexoImovel>();

            Imagens.Add(anexoImovel);
        }


    }
}

File: Corretora.Imoveis.WebApp/wwwroot/js/Imovel/imovel.js
$(document).ready(() => {
  

});


$(document).ready(() => {
    $("#btnPesquisarImovel").on('click', buscarImovelClick);

    if (_excluded)
        $("#btnPesquisarImovel").click();

});


function buscarImovelClick(e) {
    var model = new Object();
    model.Tipo = $("#cmbConsultarTipo").val();
    //model.ValorDe = +$("#txtValorImovelFiltroImovel").val();
    //model.ValorAte = +$("#txtValorImovelFiltroImovel").val();

    util.ajax.post("../Imovel/Index", model, buscarImovelClickSuccesso, buscarImovelClickError);
}

function buscarImovelClickSuccesso(HTML) {

    if (HTML) {
        $("body").empty();
        $("body").html(HTML);
        
       
    } else {
        alert("Não foram encontrados registros para sua busca .");
    }


}
function buscarImovelClickError(error) {

    alert("Erro ao buscar Imovel !");
}



function onClickCarregarImagens(e) {


    $('#btnCarregarImagens').click();
}


function onChangeImagens(e) {

    var qtdAnexos = e.files.length;


    $("#lblQtdAnexos").text(`( ${qtdAnexos} )`);
}


function onClickSalvar(e) {

    var model = new Object();
    model.Anexos = new FormData();

    model.IdImovel = +$("#txtCodigo").val();
    model.Descricao = $("#txtDescricao").val();
    model.Tipo = $("#txtTipo").val();
    model.DataPlubicacao = $("#textDataPlubicacao").val();
    model.ValorImovel = $("#txtValorImovel").val();
    model.ValorEstimadoDocumentacao = $("#txtValorEstimadoDocumentacao").val();
    model.AceitaFinanciamento = $("#txtAceitaFinanciamento").val();
    model.Observacao = $("#txtObservacao").val();
    model.Quarto = $("#txtQuarto").val();
    model.Banheiro = $("#txtBanheiro").val();
    model.Garagem = $("#txtGaragem").val();
    model.AreaUtil = $("#txtAreaUtil").val();
    model.Cep = $("#txtCep").val();
    model.Logadouro = $("#txtLogadouro").val();
    model.Numero = $("#txtNumero").val();
    model.Bairro = $("#txtBairro").val();
    model.Cidade = $("#txtCidade").val();
    model.UF = $("#txtUF").val();


    var inputFiles = document.getElementById('btnCarregarImagens');


    if (inputFiles.files.length > 0) {
        for (var indice = 0; indice < inputFiles.files.length; indice++) {
            model.Anexos.append('files', inputFiles.files.item(indice));
        }
    }


    util.ajax.post("../Imovel/Salvarimovel", model,
        (data) => {
            console.log(data);
        },

        (error) => {
            console.log(error);
        });
}

$("#meuDinheiro").maskMoney();

   


